aeon/dj_pipeline/analysis/visit_analysis.py

```diff
@@ -546,10 +545,128 @@ def make(self, key):
             self.FoodPatch.insert(food_patch_statistics)
 
 
-def _get_maintenance_periods(experiment_name, start, end):
-    # get logs from ExperimentLog
+@schema
+class VisitForagingBout(dj.Computed):
+    definition = """ # A time period spanning the time when the animal enters a food patch and moves the wheel to when it leaves the food patch
+    -> Visit
+    -> acquisition.ExperimentFoodPatch
+    bout_start: datetime(6)                    # start time of bout
+    --- 
+    bout_end: datetime(6)                      # end time of bout
+    bout_duration: float                       # (seconds)
+    wheel_distance_travelled: float            # (cm)
+    pellet_count: int                          # number of pellet trigger events
+    """
+
+    # Work on 24/7 experiments
+    key_source = (
+        Visit
+        & VisitSummary
+        & (VisitEnd & f"visit_duration > 24")
+        & f"experiment_name= 'exp0.2-r0'"
+    ) * acquisition.ExperimentFoodPatch
+
+    def make(self, key):
+        visit_start, visit_end = (VisitEnd & key).fetch1("visit_start", "visit_end")
+
+        # get in_patch timestamps
+        food_patch_description = (acquisition.ExperimentFoodPatch & key).fetch1(
+            "food_patch_description"
+        )
+        in_patch_times = np.concatenate(
+            (
+                VisitTimeDistribution.FoodPatch * acquisition.ExperimentFoodPatch & key
+            ).fetch("in_patch", order_by="visit_date")
+        )
+        maintenance_period = get_maintenance_periods(
+            key["experiment_name"], visit_start, visit_end
+        )
+        in_patch_times = filter_out_maintenance_periods(
+            pd.DataFrame(
+                [[food_patch_description]] * len(in_patch_times),
+                columns=["region"],
+                index=in_patch_times,
+            ),
+            maintenance_period,
+            visit_end,
+            True,
+        ).index.values
+
+        # get pellet data
+        chunk_keys = (
+            acquisition.Chunk
+            & f'chunk_start BETWEEN "{pd.Timestamp(visit_start).floor("H")}" AND "{visit_end}"'
+        ).fetch("KEY")
+        patch = (
+            (
+                dj.U("event_time", "event_type")
+                & (
+                    acquisition.FoodPatchEvent * acquisition.EventType
+                    & chunk_keys
```

Comment:
You may not need to search and restrict by `chunk_keys`, since you're already restricting with `event_time BETWEEN ...`